src/Dropdown/index.jsx

```diff
@@ -1,225 +1,239 @@
-      });
+    if (this.state.open) {
+      this.close();
     }
   }
 
-  handleToggleKeyDown(e) {
-    if (!this.state.open && Dropdown.isTriggerKey('OPEN_MENU', e.key)) {
-      this.toggle();
-    } else if (this.state.open && Dropdown.isTriggerKey('CLOSE_MENU', e.key)) {
-      this.toggle();
+  handleMenuKeyDown = (e) => {
+    switch (e.key) {
+      case 'ArrowUp':
+        e.preventDefault();
+        this.focusPrevious();
+        break;
+      case 'ArrowDown':
+        e.preventDefault();
+        this.focusNext();
+        break;
+      case 'Tab':
+        e.preventDefault();
+        if (e.shiftKey) {
+          this.focusPrevious();
+        } else {
+          this.focusNext();
+        }
+        break;
+      case 'Escape':
+        e.stopPropagation();
+        this.close();
+        break;
+      default:
+        break;
     }
   }
 
-  toggle() {
+  open() {
+    // adding event listener here so the user can close dropdown on click outside of the dropdown
+    document.addEventListener('click', this.handleDocumentClick, true);
     this.setState({
-      open: !this.state.open,
-      focusIndex: 0,
+      open: true,
     });
   }
 
-  generateMenuItems(menuItems) {
-    return menuItems.map((menuItem, i) => {
-      if (React.isValidElement(menuItem)) {
-        const cloneProps = {
-          ref: (item) => { this.menuItems[i] = item; },
-          className: 'dropdown-item',
-          key: i,
-          onKeyDown: this.handleMenuKeyDown,
-        };
-        return React.cloneElement(menuItem, cloneProps);
-      }
-      return (
-        <a
-          className="dropdown-item"
-          href={menuItem.href}
-          key={menuItem.href}
-          onKeyDown={this.handleMenuKeyDown}
-          ref={(item) => {
-            this.menuItems[i] = item;
-          }}
-          role="menuitem"
-        >
-          {menuItem.label}
-        </a>
-      );
+  close() {
+    document.removeEventListener('click', this.handleDocumentClick, true);
+    this.setState({
+      open: false,
     });
   }
 
+  toggle = () => {
+    if (this.state.open) {
+      this.close();
+    } else {
+      this.open();
+    }
+  }
+
   render() {
-    const { open } = this.state;
     const {
-      buttonType,
-      iconElement,
-      menuItems,
-      title,
+      buttonClassName,
+      buttonContent,
+      className,
+      children,
+      ...other
     } = this.props;
-    const hasIconElement = React.isValidElement(iconElement);
 
     return (
       <div
+        {...other}
         className={classNames(
           'dropdown',
+          className,
           {
-            show: open,
-            'has-icon': hasIconElement,
-            rounded: hasIconElement,
-            border: hasIconElement,
-            'd-flex': hasIconElement,
-            'bg-white': hasIconElement,
+            show: this.state.open,
           },
         )}
         ref={(container) => { this.container = container; }}
```

Comment:
I know we've discussed before, but can we use the new React.createRef for this ref?